File: screenshots/crop_screenshots.py
"""
Обрезка скриншотов под соотношение экрана телефона 9:19.5
для ровного отображения в рамках на лендинге.
Центральная обрезка — сохраняется середина изображения.
"""
from pathlib import Path

try:
    from PIL import Image
except ImportError:
    print("Установите Pillow: pip install Pillow")
    raise

SCREEN_RATIO = 9 / 19.5  # соотношение экрана в мокапе
SCREENSHOTS_DIR = Path(__file__).resolve().parent


def crop_to_ratio(img: Image.Image, ratio: float) -> Image.Image:
    w, h = img.size
    current_ratio = w / h
    if abs(current_ratio - ratio) < 0.001:
        return img
    if current_ratio > ratio:
        # картинка шире — обрезаем по бокам
        new_w = int(h * ratio)
        left = (w - new_w) // 2
        return img.crop((left, 0, left + new_w, h))
    else:
        # картинка выше — обрезаем сверху и снизу
        new_h = int(w / ratio)
        top = (h - new_h) // 2
        return img.crop((0, top, w, top + new_h))


def main():
    for path in sorted(SCREENSHOTS_DIR.glob("*.png")):
        if path.name.startswith("_"):
            continue
        try:
            img = Image.open(path).convert("RGB")
            cropped = crop_to_ratio(img, SCREEN_RATIO)
            cropped.save(path, "PNG", optimize=True)
            print(f"OK: {path.name} -> {cropped.size[0]}x{cropped.size[1]}")
        except Exception as e:
            print(f"Ошибка {path.name}: {e}")


if __name__ == "__main__":
    main()

File: screenshots/crop_screenshots.js
/**
 * Обрезка скриншотов под соотношение экрана 9:19.5 для ровного отображения в рамках.
 * Запуск: npx sharp-cli не подходит под crop по ratio, поэтому используем sharp в коде.
 * Запуск: node crop_screenshots.js (предварительно: npm install sharp)
 */
const fs = require('fs');
const path = require('path');

const SCREEN_RATIO = 9 / 19.5;
const DIR = __dirname;

async function run() {
  let sharp;
  try {
    sharp = require('sharp');
  } catch (_) {
    console.log('Установите sharp: npm install sharp');
    process.exit(1);
  }
  const files = fs.readdirSync(DIR).filter((f) => f.endsWith('.png') && !f.startsWith('_'));
  for (const name of files.sort()) {
    const filePath = path.join(DIR, name);
    const meta = await sharp(filePath).metadata();
    const w = meta.width;
    const h = meta.height;
    const currentRatio = w / h;
    let left = 0, top = 0, width = w, height = h;
    if (Math.abs(currentRatio - SCREEN_RATIO) < 0.001) {
      console.log('OK (без изменений):', name);
      continue;
    }
    if (currentRatio > SCREEN_RATIO) {
      width = Math.round(h * SCREEN_RATIO);
      left = Math.round((w - width) / 2);
    } else {
      height = Math.round(w / SCREEN_RATIO);
      top = Math.round((h - height) / 2);
    }
    await sharp(filePath)
      .extract({ left, top, width, height })
      .toFile(filePath + '.tmp');
    fs.renameSync(filePath + '.tmp', filePath);
    console.log('OK:', name, '->', width + 'x' + height);
  }
}

run().catch((e) => {
  console.error(e);
  process.exit(1);
});
